### ISSUE TITLE
PackageDeclarationCheck reports false positive on JEP 512 compact source files

### ISSUE THREAD

**Issue by vivek-0509** [checkstyle/checkstyle#19908]
I have read check documentation: https://checkstyle.org/checks/coding/packagedeclaration.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```
vivek@Viveks-MacBook-Air issue-pkg % mkdir -p /tmp/issue-pkg && cd /tmp/issue-pkg

cat > Hello.java << 'EOF'
void main() {
    System.out.println("Hello!");
}
EOF

vivek@Viveks-MacBook-Air issue-pkg % cat > config.xml << 'EOF'
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
  "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
  "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="PackageDeclaration"/>
  </module>
</module>
EOF

vivek@Viveks-MacBook-Air issue-pkg % RUN_LOCALE="-Duser.language=en -Duser.country=US"
java $RUN_LOCALE -jar /Users/vivek/checkstyle/target/checkstyle-13.5.0-SNAPSHOT-all.jar \
    -c config.xml Hello.java

Starting audit...
[ERROR] /private/tmp/issue-pkg/Hello.java:1:1: Missing package declaration. [PackageDeclaration]
Audit done.
Checkstyle ends with 1 errors.
```

### Describe what you expect in detail.

This file is a JEP 512 compact source file. Compact source files live in the unnamed package by definition, javac rejects an explicit package declaration inside them (see [JEP 512 §"Top-level methods and fields"](https://openjdk.org/jeps/512)). Reporting a "missing package declaration" violation against such a file is therefore a false positive, the language specification forbids the construct the check is asking the user to add.

PackageDeclarationCheck should detect that the compilation unit is a compact source file (i.e. has top-level METHOD_DEF/VARIABLE_DEF as direct children of COMPILATION_UNIT) and skip enforcement in that case.

**Comment by mahfouz72** [comment]
Can you share the CLI output of what will happen if we add the package declaration? Is it a compilation error?

**Comment by vivek-0509** [comment]
@mahfouz72 yes it's a compilation error 

```
vivek@Viveks-MacBook-Air checkstyle % cat > WithPackage.java << 'EOF'
package com.example;

void main() {
    System.out.println("Hello!");
}
EOF

vivek@Viveks-MacBook-Air checkstyle % $JAVAC WithPackage.java ; echo "exit: $?"
WithPackage.java:1: error: compact source file should not have package declaration
package com.example;
^
1 error
exit: 1
```

**Comment by romani** [comment]
fix is merged
